### PR TITLE
feat: Add Whitelist ERC20 Cli.

### PR DESCRIPTION
**Pull request by lukema95**
# Description

Add Whitelist ERC20 Cli. #1356

Closes: 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [x] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

**Comment by lukema95** [comment]
> Thanks for the contribution! Just left a few comment. We should also add an entry in `changelog.md` file for the new command

all done

File: x/crosschain/client/cli/cli_whitelist_erc20.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/spf13/cobra"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+func CmdWhitelistERC20() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "whitelist-erc20 [erc20Address] [chainID] [name] [symbol] [decimals] [gasLimit]",
+		Short: "Add a new erc20 address to whitelist",
```
Comment (kingpinXD) [inline x/crosschain/client/cli/cli_whitelist_erc20.go:16]:
```suggestion
		Short: "Add a new erc20 token to whitelist",
```

**Review by lumtis — APPROVED** [review]

**Comment by lumtis** [comment]
@lukema95 I think it is ready to be merged.
We would just need: 
- To run `make generate`
- Fix the integer overflow warning from cosmos-sec

**Comment by lukema95** [comment]
> @lukema95 I think it is ready to be merged. We would just need:
> 
> * To run `make generate`
> * Fix the integer overflow warning from cosmos-sec

all done

File: changelog.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ Getting the correct TSS address for Bitcoin now requires proviidng the Bitcoin c
 * enable zetaclients to use dynamic gas price on zetachain - enables >0 min_gas_price in feemarket module
 * add static chain data for Sepolia testnet
 * added metrics to track the burn rate of the hotkey in the telemetry server as well as prometheus
+* [1425](https://github.com/zeta-chain/node/pull/1425) add `whitelist-erc20` command
```
Comment (lumtis) [inline changelog.md:102]: Would need to be added above under unreleased

**Review by lumtis — APPROVED** [review]

**Comment by lumtis** [comment]
Seems like `make generate` has not successful

File: changelog.md (hunk 2 above)
Comment (lukema95) [reply changelog.md:102]: Will this affect the execution of `make generate`？

Comment (lumtis) [reply changelog.md:102]: No

**Comment by lukema95** [comment]
@lumtis I run `make generate`, but it doesn't do anything right now.

**Review by kingpinXD — APPROVED** [review]

**Comment by lumtis** [comment]
> @lumtis I run `make generate`, but it doesn't do anything right now.

It seems there are some differences in the proto generation. What `buf` version to you use?
```
buf --version
```
I have `1.28.0`

**Comment by lukema95** [comment]
> > @lumtis I run `make generate`, but it doesn't do anything right now.
> 
> It seems there are some differences in the proto generation. What `buf` version to you use?
> 
> ```
> buf --version
> ```
> 
> I have `1.28.0`

@lumtis I switched buf to 1.28.0 and still nothing has changed. Can you try it to see if you can generate the correct file?

**Comment by lukema95** [comment]
> > > > @lumtis I run `make generate`, but it doesn't do anything right now.
> > > 
> > > 
> > > It seems there are some differences in the proto generation. What `buf` version to you use?
> > > ```
> > > buf --version
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I have `1.28.0`
> > 
> > 
> > @lumtis I switched buf to 1.28.0 and still nothing has changed. Can you try it to see if you can generate the correct file?
> 
> This is curious, from the logs it seems you proto generation doesn't add addition lines in `common/common.pb.go` Can you maybe try adding these manually?
> 
> ```
>  type HeaderData struct {
>  	// Types that are valid to be assigned to Data:
> +	//
>  	//	*HeaderData_EthereumHeader
>  	//	*HeaderData_BitcoinHeader
>  	Data isHeaderData_Data `protobuf_oneof:"data"`
> @@ -424,6 +425,7 @@ func (*HeaderData) XXX_OneofWrappers() []interface{} {
>  
>  type Proof struct {
>  	// Types that are valid to be assigned to Proof:
> +	//
>  	//	*Proof_EthereumProof
>  	//	*Proof_BitcoinProof
>  	Proof isProof_Proof `protobuf_oneof:"proof"`
> ```

done

**Comment by lumtis** [comment]
@lukema95 in https://github.com/zeta-chain/node/pull/1425/commits/58d79f89ed1963f9e34dda896877be54b6d12c24 you modified for `HeaderData` but there is the same issue with `Proof`
Still running the CI to check if it changes something

**Comment by lukema95** [comment]
> @lukema95 in [58d79f8](https://github.com/zeta-chain/node/commit/58d79f89ed1963f9e34dda896877be54b6d12c24) you modified for `HeaderData` but there is the same issue with `Proof` Still running the CI to check if it changes something
> 
> [EDIT] Yeah we only get the error for proof now

😂It's weird. Do I need to manually fix the error for Proof again? It is not clear how many such errors there are.

**Comment by lumtis** [comment]
> > @lukema95 in [58d79f8](https://github.com/zeta-chain/node/commit/58d79f89ed1963f9e34dda896877be54b6d12c24) you modified for `HeaderData` but there is the same issue with `Proof` Still running the CI to check if it changes something
> > [EDIT] Yeah we only get the error for proof now
> 
> 😂It's weird. Do I need to manually fix the error for Proof again? It is not clear how many such errors there are.

Yeah, you can do the same thing with Proof and see if it passes

**Pull request merged**